metadata/src/main/java/org/apache/kafka/metadata/authorizer/bitmap/BitMaps.java

```diff
@@ -0,0 +1,159 @@
+        // using 0x3f (63) or compute bitIndex % 64.
+        // Note: If the index is negative the shift will be (64 - (bitIndex & 0x3f)) and
+        // this will identify an incorrect bit.
+        return 1 << bitIndex;
+    }
+
+    /**
+     * Gets the filter index for the specified bit index assuming the filter is using 64-bit longs
+     * to store bits starting at index 0.
+     *
+     * <p>The index is assumed to be positive. For a positive index the result will match
+     * {@code bitIndex / 64}.</p>
+     *
+     * <p><em>The divide is performed using bit shifts. If the input is negative the behavior
+     * is not defined.</em></p>
+     *
+     * @param bitIndex the bit index (assumed to be positive)
+     * @return the index of the bit map in an array of bit maps.
+     */
+    public static int getIntIndex(final int bitIndex) {
+        // An integer divide by 32 is equivalent to a shift of 6 bits if the integer is
+        // positive.
+        // We do not explicitly check for a negative here. Instead we use a
+        // signed shift. Any negative index will produce a negative value
+        // by sign-extension and if used as an index into an array it will throw an
+        // exception.
+        return bitIndex >> DIVIDE_BY_32;
+    }
+
+    /**
+     * Performs a modulus calculation on an unsigned long and a positive integer divisor.
+     *
+     * <p>This method computes the same result as {@link Long#remainderUnsigned(long, long)}
+     * but assumes that the divisor is an integer in the range 1 to 2<sup>31</sup> - 1 inclusive,
+     * that is a strictly positive integer size.
+     *
+     * <p><em>If the divisor is negative the behavior is not defined.</em></p>
+     *
+     * @param dividend an unsigned long value to calculate the modulus of.
+     * @param divisor the divisor for the modulus calculation, must be strictly positive.
+     * @return the remainder or modulus value.
+     * @throws ArithmeticException if the divisor is zero
+     * @see Long#remainderUnsigned(long, long)
+     */
+    public static int mod(final long dividend, final int divisor) {
+        // See Hacker's Delight (2nd ed), section 9.3.
+        // Assume divisor is positive.
+        // Divide half the unsigned number and then double the quotient result.
+        final long quotient = (dividend >>> 1) / divisor << 1;
+        final long remainder = dividend - quotient * divisor;
+        // remainder in [0, 2 * divisor)
+        return (int) (remainder >= divisor ? remainder - divisor : remainder);
+    }
+
+    /**
+     * Calculates the number of bit maps (longs) required for the numberOfBits parameter.
+     *
+     * <p><em>If the input is negative the behavior is not defined.</em></p>
+     *
+     * @param numberOfBits the number of bits to store in the array of bit maps.
+     * @return the number of bit maps necessary.
+     */
+    public static int numberOfBitMaps(final int numberOfBits) {
```

Comment:
Should we add some conditional check to avoid `numberOfBits` is a negative value?